allen-cell-animated/nucmorph-colorizer
# fix: Fixed header layout on mobile devices

Problem
=======
Closes #296, "improving header layout on mobile", as well as sprinkling in a few other fixes.

*Estimated review size: small, 15-20 minutes*

Solution
========
- Simplifies `FlexDiv` utility class and adds a prop for wrapping behavior.
- Fixes layout issue on the viewer page where the Canvas would not match screen width.
  -  EXPERIMENTALLY allows for resizing the canvas height too to maintain aspect ratio. See TODOs for possible issues with this approach.
- Header changes
  - Adds a smaller version of the AICS logo, which will be used on mobile devices.
  - Allowed elements in the header to wrap, rather than forcing the header to extend past the screen bounds.

## Type of change
* Bug fix (non-breaking change which fixes an issue)
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-297/
2. Navigate to the viewer.
3. (On Chrome) press F12 to open the developer console. Open the device toolbar.
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/d82c62c0-5bd6-4299-be93-aaf93d9ae294)

4. Click through a couple different device sizes to confirm the header and main canvas area look reasonable across them.


Screenshots (optional):
-----------------------
**Video talkthrough (🔊):**

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/fa0c1a80-b5f6-4148-8b1b-d3f7f96fec6c

![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/2144aaca-ad19-4c89-bef2-e40ac6005c64)
